src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowShuntTest.java

```diff
@@ -452,4 +452,130 @@ void testUnsupportedSharedVoltageControl() {
+                .add();
+        vl2G.getBusBreakerView().newBus()
+                .setId("b2")
+                .add();
+        vl2G.newLoad()
+                .setId("ld1")
+                .setConnectableBus("b2")
+                .setBus("b2")
+                .setP0(101)
+                .setQ0(150)
+                .add();
+        VoltageLevel vl3G = s2G.newVoltageLevel()
+                .setId("vl3")
+                .setNominalV(400)
+                .setTopologyKind(TopologyKind.BUS_BREAKER)
+                .add();
+        vl3G.getBusBreakerView().newBus()
+                .setId("b3")
+                .add();
+        ShuntCompensator shuntG = vl3G.newShuntCompensator()
+                .setId("SHUNT")
+                .setBus("b3")
+                .setConnectableBus("b3")
+                .setSectionCount(0)
+                .setVoltageRegulatorOn(true)
+                .setTargetV(393)
+                .setTargetDeadband(5.0)
+                .newNonLinearModel()
+                .beginSection()
+                .setB(1e-3)
+                .setG(1e-3)
+                .endSection()
+                .beginSection()
+                .setB(3e-3)
+                .setG(3e-3)
+                .endSection()
+                .add()
+                .add();
+        Line l1G = networkWithG.newLine()
+                .setId("l1")
+                .setVoltageLevel1("vl1")
+                .setBus1("b1")
+                .setVoltageLevel2("vl2")
+                .setBus2("b2")
+                .setR(1)
+                .setX(3)
+                .setG1(0)
+                .setG2(0)
+                .setB1(0)
+                .setB2(0)
+                .add();
+        Line l2G = networkWithG.newLine()
+                .setId("l2")
+                .setVoltageLevel1("vl3")
+                .setBus1("b3")
+                .setVoltageLevel2("vl2")
+                .setBus2("b2")
+                .setR(1)
+                .setX(3)
+                .setG1(0)
+                .setG2(0)
+                .setB1(0)
+                .setB2(0)
+                .add();
+        shuntG.setSectionCount(1);
+        LoadFlowResult result2 = loadFlowRunner.run(networkWithG, parameters);
+        assertTrue(result2.isOk());
+        assertActivePowerEquals(101.366, l1G.getTerminal(Branch.Side.ONE));
+        assertActivePowerEquals(-101.299, l1G.getTerminal(Branch.Side.TWO));
+        assertReactivePowerEquals(-1.398, l1G.getTerminal(Branch.Side.ONE));
+        assertReactivePowerEquals(1.600, l1G.getTerminal(Branch.Side.TWO));
+        assertActivePowerEquals(-152.513, l2G.getTerminal(Branch.Side.ONE));
+        assertActivePowerEquals(152.818, l2G.getTerminal(Branch.Side.TWO));
+        assertReactivePowerEquals(152.514, l2G.getTerminal(Branch.Side.ONE));
+        assertReactivePowerEquals(-151.599, l2G.getTerminal(Branch.Side.TWO));
+        //assertActivePowerEquals(152.513, shuntG.getTerminal());
```

Comment:
I have tried to fix that and I fall in:
````
        if (!Double.isNaN(p) && connectable.getType() == IdentifiableType.SHUNT_COMPENSATOR) {
            throw new ValidationException(connectable, "cannot set active power on a shunt compensator");
        }
````

Reply:
we have to open a fix PR on core side before next release to drop this abusive check.